src/app/Dashboard/DashboardLayoutConfig.tsx

```diff
@@ -0,0 +1,398 @@
+      onFocus();
+    },
+    [dispatch, onFocus, dashboardConfigs]
+  );
+
+  const onActionClick = React.useCallback(
+    (ev: React.MouseEvent<HTMLButtonElement, MouseEvent>, itemId: string, actionId: string) => {
+      ev.stopPropagation(); // prevent the dropdown from closing
+      if (actionId === 'rename') {
+        handleRenameLayout(itemId);
+      } else if (actionId === 'delete') {
+        handleDeleteButton(ev, itemId);
+      } else if (actionId === 'fav') {
+        handleFavoriteLayout(itemId);
+      }
+    },
+    [handleRenameLayout, handleDeleteButton, handleFavoriteLayout]
+  );
+
+  const onToggle = React.useCallback(
+    (_ev: React.MouseEvent<Element, MouseEvent>) => {
+      setIsSelectorOpen((open) => !open);
+    },
+    [setIsSelectorOpen]
+  );
+
+  const newButton = React.useMemo(
+    () => (
+      <Button
+        key="new"
+        variant="primary"
+        aria-label={t('DashboardLayoutConfig.NEW.LABEL')}
+        onClick={handleCreateModalOpen}
+        icon={<PlusCircleIcon />}
+      >
+        {t('NEW', { ns: 'common' })}
+      </Button>
+    ),
+    [t, handleCreateModalOpen]
+  );
+
+  const renameButton = React.useMemo(
+    () => (
+      <Button
+        key="rename"
+        variant="plain"
+        isAriaDisabled={currLayout.name === DEFAULT_DASHBOARD_NAME}
+        aria-label={t('DashboardLayoutConfig.RENAME.LABEL')}
+        onClick={() => handleRenameLayout(currLayout.name)}
+        icon={<PencilAltIcon />}
+      />
+    ),
+    [t, handleRenameLayout, currLayout.name]
+  );
+
+  const uploadButton = React.useMemo(
+    () => (
+      <Button
+        key="upload"
+        variant="secondary"
+        aria-label={t('DashboardLayoutConfig.UPLOAD.LABEL')}
+        onClick={handleUploadModalOpen}
+        icon={<UploadIcon />}
+      >
+        {t('UPLOAD', { ns: 'common' })}
+      </Button>
+    ),
+    [t, handleUploadModalOpen]
+  );
+
+  const downloadButton = React.useMemo(
+    () => (
+      <Button
+        key="download"
+        variant="secondary"
+        aria-label={t('DashboardLayoutConfig.DOWNLOAD.LABEL')}
+        onClick={handleDownloadLayout}
+        icon={<DownloadIcon />}
+      >
+        {t('DOWNLOAD', { ns: 'common' })}
+      </Button>
+    ),
+    [t, handleDownloadLayout]
+  );
+
+  const deleteButton = React.useMemo(
+    () => (
+      <Button
+        ref={deleteRef}
+        key="delete"
+        variant="danger"
+        isAriaDisabled={currLayout.name === DEFAULT_DASHBOARD_NAME}
+        aria-label={t('DashboardLayoutConfig.DELETE.LABEL')}
+        onClick={(ev) => handleDeleteButton(ev, currLayout.name)}
+        icon={<TrashIcon />}
+      >
+        {t('DELETE', { ns: 'common' })}
+      </Button>
+    ),
+    [t, handleDeleteButton, currLayout.name]
+  );
+
+  const menuGroups = React.useCallback(
+    (label: string, favoriteGroup: boolean) => {
+      const filter = favoriteGroup ? (l: DashboardLayout) => l.favorite : () => true;
+      return (
```

Comment:
We probably should hide `Favorite` group if there is none. For example, here we have an empty group above.

![Screenshot from 2023-03-19 23-53-19](https://user-images.githubusercontent.com/68053619/226242675-f242332f-c0c5-4857-80ec-c7ffe288c8ba.png)

I think we could extract the filtering out and check children array length?

Reply:
Just to be consistent with target selector :))